File: HW/HW1.py
def is_palindrome(s):
    # Convert the input string to lowercase
    lowercase_string = s.lower()
    
    # Remove all non-alphanumeric characters from the string
    alphanumeric_string = ''.join(char for char in lowercase_string if char.isalnum())
    
    # Check if the modified string is a palindrome
    return alphanumeric_string == alphanumeric_string[::-1]

# Testing function
input_string = "A man, a plan, a canal: Panama"
is_palindrome_result = is_palindrome(input_string)
print("Is the string a palindrome?", is_palindrome_result)

File: HW/HW1.java
public class CheckPalindrome {
    public static void isPalindrome(String s) {
        // Convert input string to lowercase
        String lowercaseString = s.toLowerCase();
        
        // Remove non-alphanumeric
        StringBuilder alphanumericString = new StringBuilder();
        for (char c : lowercaseString.toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                alphanumericString.append(c);
            }
        }
        
        // Check string
        boolean palindromeCheck = alphanumericString.toString().equals(alphanumericString.reverse().toString());
        
        // Print the result
        if (palindromeCheck) {
            System.out.println("Is the string a palindrome? Yes.");
        } else {
            System.out.println("Is the string a palindrome? No.");
        }
    }

    public static void main(String[] args) {
        String inputString = "Murder for a jar of red rum.";
        isPalindrome(inputString);
    }
}

